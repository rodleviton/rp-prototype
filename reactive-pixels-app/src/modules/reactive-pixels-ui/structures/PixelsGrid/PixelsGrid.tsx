import classnames from "classnames";
import { css } from "emotion";
import * as React from "react";
import { Grid } from "../../components/Grid";
import { IBaseTheme, withStyles } from "../../theme";

interface IClasses {
  root: string;
}

interface IProps {
  classes: IClasses;
  className?: string;
}

const styles = (theme: IBaseTheme): IClasses => {
  const { breakpoints } = theme;
  const GUTTER = 20;
  const ITEM_WIDTH = 340 + GUTTER * 2;

  return {
    root: css({
      margin: "0 auto",
      width: ITEM_WIDTH,

      [`@media (min-width: ${breakpoints.medium}px)`]: {
        width: ITEM_WIDTH * 2
      },
      [`@media (min-width: ${breakpoints.xl}px)`]: {
        width: ITEM_WIDTH * 3
      },
      [`@media (min-width: ${breakpoints.xxl}px)`]: {
        width: ITEM_WIDTH * 4
      }
    })
  };
};

/**
 * PixelsGrid
 *
 * @param {string} [className=""] - Extend css classes to override default styling.
 * @example
 * <PixelsGrid />
 */
export class PixelsGrid extends React.PureComponent<IProps> {
  public render() {
    const { classes, className, children, ...otherProps } = this.props;

    return (
      <Grid
        flexWrap="wrap"
        className={classnames(classes.root, className)}
        {...otherProps}
      >
        {children}
      </Grid>
    );
  }
}

export default withStyles(styles)(PixelsGrid);
