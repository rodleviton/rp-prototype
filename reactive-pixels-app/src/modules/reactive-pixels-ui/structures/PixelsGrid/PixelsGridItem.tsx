import classnames from "classnames";
import { css } from "emotion";
import * as React from "react";
import { Box } from "../../components/Box";
import { IBaseTheme, withStyles } from "../../theme";

interface IClasses {
  root: string;
}

interface IProps {
  classes: IClasses;
  className?: string;
}

const styles = (theme: IBaseTheme): IClasses => {
  return {
    root: css({
      margin: 20
    })
  };
};

/**
 * PixelsGridItem
 *
 * @param {string} [className=""] - Extend css classes to override default styling.
 * @example
 * <PixelsGridItem />
 */
export class PixelsGridItem extends React.PureComponent<IProps> {
  public render() {
    const { classes, className, children, ...otherProps } = this.props;

    return (
      <Box className={classnames(classes.root, className)} {...otherProps}>
        {children}
      </Box>
    );
  }
}

export default withStyles(styles)(PixelsGridItem);
