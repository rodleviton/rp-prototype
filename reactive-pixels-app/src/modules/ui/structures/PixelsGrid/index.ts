export { default as PixelsGrid } from "./PixelsGrid";
export { default as PixelsGridItem } from "./PixelsGridItem";
