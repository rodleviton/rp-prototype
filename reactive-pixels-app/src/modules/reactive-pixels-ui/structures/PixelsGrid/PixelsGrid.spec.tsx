import * as React from "react";
import { render } from "react-testing-library";
import WithRouter from "../../utils/test/WithRouter";
import WithTheme from "../../utils/test/WithTheme";
import { PixelsGrid } from "../PixelsGrid";

describe("<PixelsGrid />", () => {
  it("should render <PixelsGrid /> component", () => {
    const { container } = render(
      <WithRouter>
        <WithTheme>
          <PixelsGrid />
        </WithTheme>
      </WithRouter>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
