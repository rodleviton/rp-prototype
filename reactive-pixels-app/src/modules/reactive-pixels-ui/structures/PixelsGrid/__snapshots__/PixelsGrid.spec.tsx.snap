// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PixelsGrid /> should render <PixelsGrid /> component 1`] = `
<div
  class="css-0 css-11801zi sc-bwzfXH liBCIH sc-bdVaJa bDWFJH"
/>
`;
